cmd/trace-agent/config/config.go

```diff
@@ -221,8 +221,8 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 	if coreconfig.Datadog.IsSet("apm_config.errors_per_second") {
 		c.ErrorTPS = coreconfig.Datadog.GetFloat64("apm_config.errors_per_second")
 	}
-	if coreconfig.Datadog.IsSet("apm_config.disable_rare_sampler") {
```

Comment:
Can this just be renamed to apm_config.rare_sampler and have it default to true?

Also, since we're already renaming this, maybe we should warn people if they are using `apm_config.disable_rare_sampler` that they should migrate over to using `apm_config.rare_sampler` if they wish to enable to rare sampler (noting that it's off by default).

Same thing for the environment variable, though I think that could go in the `load` function of this file.

Reply:
Since these options can hypothetically be any valid yaml I used `enable` to make it clear that it should be a boolean value, although I can change to just `rare_sampler` if you think that's better?

There is a warning for unknown config options so a customer would get a log like "We don't recognized apm_config.disable_rare_sampler" but I agree a specific warning here would be nice so I'll add one.

Reply:
I've updated the PR with a warning (and checking just apm_config.disable_rare_sampler is sufficient since the envvar gets translated into the config before that point)